lib/string.go

```diff
@@ -0,0 +1,23 @@
+package lib
+
+import (
+	"strings"
+)
+
+// StrContains checks if a list contains a string
+func StrContains(l []string, s string) bool {
+	for _, v := range l {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func ToLowerList(l []string) []string {
```

Comment:
There is actually one non-test place where this code was used (`command/agent/config.go`) and it was easier to shoot first, ask questions later since the refactor was mostly mechanical and it crossed runtime and unit tests.

```
% egrep -r StrContains *
command/agent/config.go:		if !lib.StrContains(allowedKeys, field) {
command/agent/config_test.go:	if !lib.StrContains(serv.Tags, "master") {
consul/acl_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/catalog_endpoint_test.go:	if !lib.StrContains(services["db"].Tags, "primary") || services["db"].Port != 5000 {
consul/fsm_test.go:	if !lib.StrContains(fooSrv.Services["db"].Tags, "primary") {
consul/health_endpoint_test.go:	if !lib.StrContains(nodes[0].Service.Tags, "slave") {
consul/health_endpoint_test.go:	if !lib.StrContains(nodes[1].Service.Tags, "master") {
consul/internal_endpoint_test.go:	if !lib.StrContains(nodes[0].Services[0].Tags, "master") {
consul/internal_endpoint_test.go:			if !lib.StrContains(node.Services[0].Tags, "master") {
consul/internal_endpoint_test.go:			if !lib.StrContains(node.Services[0].Tags, "slave") {
consul/session_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/session_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/session_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/session_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/session_endpoint_test.go:			if !lib.StrContains(ids, s.ID) {
consul/session_endpoint_test.go:		if !lib.StrContains(ids, s.ID) {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[0].ServiceTags, "slave") {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[1].ServiceTags, "slave") {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[2].ServiceTags, "master") {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[0].ServiceTags, "master") {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[0].ServiceTags, "master") {
consul/state/state_store_test.go:	if !lib.StrContains(nodes[0].ServiceTags, "dev") {
lib/string.go:// StrContains checks if a list contains a string
lib/string.go:func StrContains(l []string, s string) bool {
lib/string_test.go:func TestStrContains(t *testing.T) {
lib/string_test.go:	if !StrContains(l, "b") {
lib/string_test.go:	if StrContains(l, "d") {
```